app/models/assignment_form.rb

```diff
@@ -115,30 +115,38 @@ def add_to_delayed_queue
     duedates = AssignmentDueDate.where(parent_id: @assignment.id)
     duedates.each do |due_date|
       deadline_type = DeadlineType.find(due_date.deadline_type_id).name
-      due_at = due_date.due_at.to_s(:db)
-      Time.parse(due_at)
-      due_at = Time.parse(due_at)
-      mi = find_min_from_now(due_at)
-      diff = mi - due_date.threshold * 60
+      diff, mi = get_diff_from_now(due_date)
```

Comment:
Variable names in original code is not reasonable, please rename these variables, like `diff`, `mi`.